src/EditorFeatures/Core/EditAndContinue/EditAndContinueLanguageService.cs

```diff
@@ -102,12 +97,18 @@ internal void Disable(Exception e)
             .ReportNonFatalErrorAsync().CompletesAsyncOperation(token);
     }
 
+    private void UpdateApplyChangesDiagnostics(ImmutableArray<DiagnosticData> diagnostics)
+    {
+        sessionState.ApplyChangesDiagnostics = diagnostics;
+        diagnosticRefresher.RequestWorkspaceRefresh();
```

Comment:
Are tehre threadsafety concerns here around this mutation?

Reply:
I don't think so, it's an atomic reference field update (underlying array of the ImmutableArray).